### PR TITLE
docs/update add GCE custom image deployment type

### PR DESCRIPTION
**Pull request by abeatrix**
Close: https://github.com/orgs/sourcegraph/projects/290

This PR adds GCE custom image deployment type to docs.

Pipeline for creating GCE images automatically can be found in https://github.com/sourcegraph/deploy/commit/4e8f722ceccc41a829eec1d07827868c1e42a2e3

All GCE image files are stored in the Storage bucket under `Sourcegraph AMI project`

These images pull the latest helm charts when users first launch their instance initially.

Here is a loom video of setting up the image instance following the instruction in the PR:
https://www.loom.com/share/4f98cb9a063b464699a9323e593de93e

Upgrade walkthrough using the instruction in the PR:
https://www.loom.com/share/cd1c19fa97f34e31bab411649974e71f

**Both the initial deployment and upgrade took less than 3 mins to stand up after the VM has been stood up**

### Docs Update 1

Added GCE Images section with detailed instructions on deploying Sourcegraph

image

image

### Docs Update 2

Also replaced some of the listed links with button styles. See example screenshots below:

Homepage New:
Screen Shot 2022-10-21 at 8 57 14 AM

Homepage Old:
image

Docker Compose index page - New:
image

Docker Compose index page - Old:
image

####

## Test plan

run `yarn docsite:serve` to see build locally

### CODE REVIEW

File: doc/admin/deploy/images/gce.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+---
+
+## Deploy Sourcegraph
+
+#### Step 1: Upload the instance image to your GCP storage bucket
+
+1. In the [instance size chart](#instance-size-chart) below, download the image (`tar.gz`) file that matches your deployment size.
+2. Upload the tar.gz file to your GCP Storage bucket
+3. Copy the **URL**
```
Comment (jjeffwarner) [inline doc/admin/deploy/images/gce.md]: From the three vertical dots, a Copy Authenticated URL and pasting it into the Cloud Storage File didn't work. Also a browser copy on the .tar.gz file in the bucket window didn't work either. I ended up using the `BROWSE` button in the Cloud Storage file input field to find it.

Comment (abeatrix) [reply doc/admin/deploy/images/gce.md]: Updated steps to reflect this, thanks for catching that @jjeffwarner !

File: doc/admin/deploy/index.md [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
+- For most customers, we recommend Sourcegraph Cloud, managed entirely by Sourcegraph.
+- For customers who want to self-host, we recommend one of the single-node deployment options.
+- For enterprise customers that require a multi-node, self-hosted deployment, we offer a Kubernetes option. We strongly encourage you to get in touch by emails (sales@sourcegraph.com) if you pursue this option.
+- If you are short on time and looking for a quick way to test Sourcegraph locally, consider running Sourcegraph via our [Docker Single Container](docker-single-container/index.md).
 
 ## Deployment types
 
-Each of the deployment types listed below provides a different level of capability. As mentioned previously, base your deployment type on the needs of your business. However, you should also consider the technical expertise available for your deployment. The sections below provide more detailed recommendations for each deployment type.
+To start, you will need to decide your on deployment method, including Kubernetes with or without Helm, as they are noninterchangeable it. In short, you **cannot** change your deployment type of a running instance.
```
Comment (jdpleiness) [inline doc/admin/deploy/index.md:26]:
```suggestion
To start, you will need to decide your on deployment method, including Kubernetes with or without Helm, as they are noninterchangeable. In short, you **cannot** change your deployment type of a running instance.
```

File: doc/admin/deploy/index.md [hunk 3]
```diff
@@ -30,13 +64,33 @@ Helm provides a simple mechanism for deployment customizations, as well as a muc
...
 If you are unable to use Helm to deploy, but still want to use Kubernetes, follow our [Kubernetes deployment documentation](kubernetes/index.md). This path will require advanced knowledge of Kubernetes. For teams without the ability to support this, please speak to your Sourcegraph contact about using Docker Compose instead.
 
+---
+
 ## Reference repositories
 
 Sourcegraph provides reference repositories with branches corresponding to the version of Sourcegraph you wish to deploy. The reference repository contains everything you need to spin up and configure your instance depending on your deployment type, which also assists in your upgrade process going forward.
 
-Before starting, you will need to decide your deployment type, including if you would like to use Kubernetes with Helm (vs. a more manual customization path). In the case of Kubernetes once you choose Helm (or not), it **can't be changed afterwards**.
+For more information, follow the install and configuration docs for your specific deployment type.
```
Comment (jdpleiness) [inline doc/admin/deploy/index.md:73]:
```suggestion
For more information, follow the installation and configuration docs for your specific deployment type.
```

File: doc/admin/deploy/index.md [hunk 4]
```diff
@@ -30,13 +64,33 @@ Helm provides a simple mechanism for deployment customizations, as well as a muc
...
+
+## Configuration
+
+Configuration at the deployment level focuses on ensuring your Sourcegraph runs optimally, based on the size of your repositories and the number of users. Configuration options will vary based on the type of deployment you choose. Consult the specific configuration deployment sections for additional information.
```
Comment (jdpleiness) [inline doc/admin/deploy/index.md:77]:
```suggestion
Configuration at the deployment level focuses on ensuring your Sourcegraph deployment runs optimally, based on the size of your repositories and the number of users. Configuration options will vary based on the type of deployment you choose. Consult the specific configuration deployment sections for additional information.
```

File: doc/admin/deploy/machine-images/gce.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+---
+
+## Deploy Sourcegraph
+
+#### Step 1: Upload the Sourcegraph instance image to your Google Cloud Storage
+
+<img class="screenshot" src="./assets/gcp-bucket.png"/>
+
+1. In the [instance size chart](#instance-size-chart) below, download the image (`tar.gz`) file that matches your deployment size
+2. Nevigate to [Google Cloud Storage](https://console.cloud.google.com/storage) and select a project where you will store the image and deploy the instance in
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/gce.md:28]:
```suggestion
2. Navigate to [Google Cloud Storage](https://console.cloud.google.com/storage) and select a project where you will store the image and deploy the instance in
```

File: doc/admin/deploy/machine-images/gce.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+2. Click **CREATE** to create your custom image of Sourcegraph
+
+#### Step 3: Launch a new instance
+
+<img class="screenshot" src="./assets/gcp-launch.png"/>
+
+1. Create a new VM instance from your [GCE Console](https://console.cloud.google.com/compute)
+2. Name the instance
+3. Select a **machine type** (see [instance size chart](#instance-size-chart))
+4. Under the **Boot disk** section, click **CHANGE** and nevigate to the **CUSTOM IMAGES** tab
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/gce.md:51]:
```suggestion
4. Under the **Boot disk** section, click **CHANGE** and navigate to the **CUSTOM IMAGES** tab
```

File: doc/admin/deploy/machine-images/gce.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+**sourcegraph** is the default username for all GCE image instances.
+
+To SSH into the instance as user `sourcegraph`, add the username in front of your instance name. For example:
+
+```bash
+# sourcegraph is the username
+$ gcloud compute ssh sourcegraph@your-instance-name --zone=us-central1-a
+```
+Your can also switch user in the terminal with the command below:
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/gce.md:76]:
```suggestion
Your can also switch users in the terminal with the command below:
```

File: doc/admin/deploy/machine-images/gce.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+$ sudo su sourcegraph
+```
+
+---
+
+## Instance size chart
+
+Select a deployment image according to the number of users and repositories you have using this table. 
+
+If you fall between two sizes, choose the larger of the two. For example, if you have 8,000 users with 80,000 repositories, your instance size would be **M**. If you have 1,000 users with 80,000 repositories, you should still go with size **M**.
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/gce.md:88]: Wouldn't a deployment with 8,000 users and 80,000 repositories choose a **L** according to this chart?

File: doc/admin/deploy/machine-images/gce.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+#### Option 2
+
+##### Step 1: Stop the current instance
+
+<img class="screenshot" src="./assets/gcp-edit.png"/>
+
+1. *Stop* your current Sourcegraph GCP Image instance from the [GCE Console](https://console.cloud.google.com/compute)
+2. Nevigate to your instance console after your instance is stopped
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/gce.md:141]:
```suggestion
2. Navigate to your instance console after your instance is stopped
```

**Review by jdpleiness — CHANGES_REQUESTED** [review]
Awesome work 😄 

I am wondering about the clarity of our instance sizing chart however. Perhaps we should add ranges to it? For example: 
|                  | **XS**     | **S**       | **M**       | **L**       | **XL**      |
|------------------|------------|-------------|-------------|-------------|-------------|
| **Users**        | 500  - 1,000       | 1,000 - 5,000      | 5,000 - 10,000      | 10,000  - 20,000    | 20,000+      |
| **Repositories** | 5,000 - 10,000     | 10,000 - 50,000    | 50,000 - 100,000     | 100,000 - 250,000    | 250,000+     |
| **Series**       | N2         | N2          | N2          | N2          | N2          |

File: doc/admin/deploy/machine-images/gce.md (hunk 8 above)
Comment (abeatrix) [reply doc/admin/deploy/machine-images/gce.md:88]: Ohhh nice catch!!!!

**Review by jdpleiness — APPROVED** [review]

**Comment by abeatrix** [comment]
@jdpleiness Good point! I have updated the range to `<=` to keep the table within page width as discussed 😄 Thank you so much!!

image

**Pull request merged**